src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs

```diff
@@ -128,6 +128,32 @@ private static int FindIndexOfAny(string format, char[] chars, int startIndex, i
         }
 
         public string Format(object?[]? values)
+        {
+            object?[]? formattedValues = values;
+
+            if (values != null)
```

Comment:
IMO there should be a comment here explaining why this is happening

Reply:
Explaining why it does an allocation of a new array in case the value would be changed? Yes, I'll add a comment about that!